framework/web/UrlRule.php

```diff
@@ -213,6 +218,95 @@ public function init()
         if (!empty($this->_routeParams)) {
             $this->_routeRule = '#^' . strtr($this->route, $tr2) . '$#u';
         }
+
+        $this->initCanonizer();
+    }
+
+    /**
+     * Initializes canonizer configuration
+     * @throws \Exception
+     */
+    protected function initCanonizer() {
+        $canonize = $this->canonize;
+
+        $default = [
+            // whether to collapse multiple slashes into one
+            'collapse-slashes' => false,
+            // whether trailing slash required
+            // true: yes
+            // false: no
+            // null: leave as is
+            'trailing-slash' => null,
+            // what action to perform if pathInfo was changed during canonization
+            // redirect: redirect to canonical path
+            // 404: throw NotFound exception
+            // none: no action, just route by canonized path
+            'action' => 'none'
+        ];
```

Comment:
It can be set via separate options in rule config, i. e.:

```
        'ruleConfig' => [
            'class' => 'yii\web\UrlRule',
            'canonize' => [
                'trailing-slash' => null, // ignore whether trailing slash exists or not
                'collapse-slashes' => false, // do not warry about multiple slashes
                'action' => 'none' // do nothing
            ]
        ],
```

Or it can be set via shortcut:

```
        'ruleConfig' => [
            'class' => 'yii\web\UrlRule',
            'canonize' => 'trailing-slash' // expands to { trailing-slash: true, collapse-slashes: true, action: redirect }
        ],
```

Is it OK? If not, could you pease give an example of how it should look like in configuration?

Currently supported shortcuts: 'trailing-shash', 'no-trailing-shash', 'disabled'.

By default if  ```canonize``` not set, behavior is not changed.